components/addressmanager/src/lib.rs

```diff
@@ -147,6 +172,34 @@ impl AddressManager {
             normalized_p2p_listen_address.into()
         };
 
+        let mut index = 0;
+        let already_in_use = loop {
```

Comment:
Add a comment that explains why this code is needed in addition to handling the error in `AddPortError::PortInUse`